dataprofiler/data_readers/data_utils.py

```diff
@@ -439,52 +441,117 @@ def read_csv_df(
+    if n_rows > sample_nrows:
+        select = np.array([False] * n_rows)
+        select[random.sample(range(n_rows), sample_nrows)] = True
+    else:
+        select = np.array([True] * n_rows)
+    out = table.filter(select).to_pandas()
 
-        # Convert all the unicode columns to utf-8
-        types = data_row_df.apply(
-            lambda x: pd.api.types.infer_dtype(x.values, skipna=True)
-        )
+    # Convert all the unicode columns to utf-8
+    types = out.apply(lambda x: pd.api.types.infer_dtype(x.values, skipna=True))
 
-        mixed_and_unicode_cols = types[types == "unicode"].index.union(
-            types[types == "mixed"].index
+    mixed_and_unicode_cols = types[types == "unicode"].index.union(
+        types[types == "mixed"].index
+    )
+
+    for col in mixed_and_unicode_cols:
+        out[col] = out[col].apply(
+            lambda x: x.encode("utf-8").strip() if isinstance(x, str) else x
+        )
+        out[col] = out[col].apply(
+            lambda x: x.decode("utf-8").strip() if isinstance(x, bytes) else x
         )
+    original_df_dtypes = out.dtypes
+    if read_in_string:
+        out = out.astype(str)
+
+    return out, original_df_dtypes
+
 
-        for col in mixed_and_unicode_cols:
-            data_row_df[col] = data_row_df[col].apply(
-                lambda x: x.encode("utf-8").strip() if isinstance(x, str) else x
+def read_parquet_df(
+    file_path: str,
+    sample_nrows: Optional[int] = None,
+    selected_columns: Optional[List[str]] = None,
+    read_in_string: bool = False,
+) -> Tuple[pd.DataFrame, pd.Series]:
+    """
+    Return an iterator that returns one row group each time.
+
+    :param file_path: path to the Parquet file.
+    :type file_path: str
+    :return:
+    :rtype: Iterator(pd.DataFrame)
+    """
+    if sample_nrows is None:
+        parquet_file = pq.ParquetFile(file_path)
+        data = pd.DataFrame()
+        for i in range(parquet_file.num_row_groups):
+
+            data_row_df = parquet_file.read_row_group(i).to_pandas()
+
+            # Convert all the unicode columns to utf-8
+            types = data_row_df.apply(
+                lambda x: pd.api.types.infer_dtype(x.values, skipna=True)
             )
-            data_row_df[col] = data_row_df[col].apply(
-                lambda x: x.decode("utf-8").strip() if isinstance(x, bytes) else x
+
+            mixed_and_unicode_cols = types[types == "unicode"].index.union(
+                types[types == "mixed"].index
             )
 
-        if selected_columns:
-            data_row_df = data_row_df[selected_columns]
+            for col in mixed_and_unicode_cols:
+                data_row_df[col] = data_row_df[col].apply(
+                    lambda x: x.encode("utf-8").strip() if isinstance(x, str) else x
+                )
+                data_row_df[col] = data_row_df[col].apply(
+                    lambda x: x.decode("utf-8").strip() if isinstance(x, bytes) else x
+                )
```

Comment:
lots of repeated code from the above function, I wonder if this can be pulled out into it's own function. That way we can just do a function call here and we don't need to repeat code. Just a thought, not a hard ask.

Reply:
yeah, I agree @micdavis -- I did just take a look at `main` code in `data_utils.py` and its interesting because that was part of the original release so we do know it works, but also see opportunity to improve. 

Since it was "original release" code, I'm wondering to hold off changing in this PR actually. Thoughts? Follow-up PR @micdavis  (CC @menglinw)

Reply:
pulled out as a function `convert_unicode_col_to_utf8` to reduce repeating code,. If we want to holding the change, ready to change it back.